nixos/modules/services/web-servers/caddy/default.nix

```diff
@@ -267,6 +271,15 @@ in
       '';
     };
 
+    enableReload = mkOption {
+      default = false;
```

Comment:
[The docs](https://caddyserver.com/docs/command-line#caddy-reload) don’t mention any setting that couldn’t be handled by the reload command. However the reload command needs the admin endpoint to be enabled:

> Because this command uses the API, the admin endpoint must not be disabled.

Since [the admin endpoint can be disabled](https://caddyserver.com/docs/json/admin/) the reload command might not work in all environments. But we could still set the option by default and allow disabling it.

Reply:
Will there be an error reported by `nixos-rebuild` if this option is set and the admin API is disabled?  If so, I recommend enabling it by default.

Reply:
I just tested it and yes, there’s the usual systemd error:

```
reloading the following units: caddy.service
Job for caddy.service failed.
See "systemctl status caddy.service" and "journalctl -xeu caddy.service" for details.
warning: error(s) occurred while switching to the new configuration
```

Note the API is enabled by default so you have to disable it explicitly for this to fail. In the current version of this module this is already a "problem" since it makes `systemctl reload caddy` unusable. I’m not sure if we should do something about it (except maybe mentioning it in the `enableReload` docs).

Reply:
What about enabling this by default and mentioning the need of admin API in the description and maybe also in the release note?

Reply:
I added a note about the admin API in the description & in the release notes. I’m not sure about enabling this option by default because of the case of the `grace_period` setting mentioned by @IndeedNotJames, which could in theory delay the reload indefinitely.

Should we enable it by default, the mention of the `grace_period` setting should be in the release notes (maybe in the backwards incompatibilities section?).

Reply:
> Should we enable it by default, the mention of the grace_period setting should be in the release notes (maybe in the backwards incompatibilities section?).

I agree.

I am slightly prefer to enable it by default in this pr, but it's also fine to do that in another pr if needed.

This is not a blocker to get my approval.

Reply:
I’ve enabled it by default and added a note to the backwards incompatibilities section of the release notes.